tests/testthat/test-linter_tags.R

```diff
@@ -27,3 +27,101 @@ test_that("available_linters matches the set of linters available from lintr", {
   # ensure that all _linter objects in our namespace are also exported
   expect_setequal(all_linters, grep("_linter$", getNamespaceExports("lintr"), value = TRUE))
 })
+
+# See the roxygen helpers in R/linter_tags.R for the code used to generate the docs.
+#   This test helps ensure the documentation is up to date with the available_linters() database
+test_that("lintr help files are up to date", {
+  helper_db_dir <- system.file("help", package = "lintr")
+  skip_if_not(dir.exists(helper_db_dir))
+  skip_if_not(file.exists(file.path(helper_db_dir, "lintr.rdb")))
+  skip_if_not(file.exists(file.path(helper_db_dir, "lintr.rdx")))
+
+  help_env <- new.env(parent = topenv())
+  lazyLoad(file.path(helper_db_dir, "lintr"), help_env)
+
+  lintr_db <- available_linters()
+  lintr_db$package <- NULL
+  lintr_db$tags <- lapply(lintr_db$tags, sort)
+
+  expect_true(exists("linters", envir = help_env), info = "?linters exists")
+  # objects in help_env are class Rd, see ?as.character.Rd (part of 'tools')
+  linter_help_text <- paste(as.character(help_env$linters), collapse = "")
+
+  # Test two things about ?linters
+  #   (1) the complete list of linters and tags matches that in available_linters()
+  #   (2) the tabulation of tags & corresponding count of linters matches that in available_linters()
+
+  # Rd markup for items looks like \item{\code{\link{...}} (tags: ...)}
+  help_linters <- rex::re_matches(
+    linter_help_text,
+    rex::rex(
+      "\\item{\\code{\\link{",
+      capture(some_of(letter, number, "_", "."), name = "linter"),
+      "}} (tags: ",
+      capture(some_of(letter, "_", ",", " "), name = "tags"),
+      ")}"
+    ),
+    global = TRUE
+  )[[1L]]
+  help_linters$tags <- lapply(strsplit(help_linters$tags, ", ", fixed = TRUE), sort)
+
+
+  # (1) from above
+  expect_identical(
+    help_linters[order(help_linters$linter), ],
+    lintr_db[order(lintr_db$linter), ],
+    info = "Database implied by ?linters is the same as is available_linters()"
+  )
+
+  db_tag_table <- as.data.frame(
+    table(tag = unlist(lintr_db$tags)),
```

Comment:
Curious: What will happen if we accidentally provide a tag twice in `linters.csv`?